test/dependency-manager/external.test.ts

```diff
@@ -0,0 +1,282 @@
+import { ServiceNode } from '../../src/dependency-manager/src';
+import { ExternalNode } from '../../src/dependency-manager/src/graph/node/external';
+
+describe('external nodes', function () {
+  beforeEach(async () => {
+    // Stub the logger
+    sinon.replace(Build.prototype, 'log', sinon.stub());
+    moxios.install();
+  });
+
+  afterEach(function () {
+    // Restore stubs
+    sinon.restore();
+    // Restore fs
+    mock_fs.restore();
+    moxios.uninstall();
+  });
+
+  it('simple external service', async () => {
+    const frontend_config = {
+      name: 'architect/frontend',
+      interfaces: {
+        app: 8080
+      }
+    };
+
+    moxios.stubRequest(`/accounts/architect/services/frontend/versions/v1`, {
+      status: 200,
+      response: { tag: 'v1', config: frontend_config, service: { url: 'architect/frontend:v1' } }
+    });
+
+    const env_config = {
+      services: {
+        'architect/frontend:v1': {
+          interfaces: {
+            app: {
+              host: 'app.localhost',
+              port: 80
+            }
+          }
+        }
+      }
+    };
+
+    mock_fs({
+      '/stack/src/frontend/architect.json': JSON.stringify(frontend_config),
+      '/stack/arc.env.json': JSON.stringify(env_config),
+    });
+
+    const manager = await LocalDependencyManager.createFromPath(axios.create(), '/stack/arc.env.json', undefined, true);
+    const graph = manager.graph;
+    expect(graph.nodes).length(1);
+    expect(graph.nodes[0]).instanceOf(ExternalNode);
+    expect(graph.nodes[0].interfaces.app.host).eq('app.localhost');
+    expect(graph.nodes[0].interfaces.app.port).eq(80);
+    expect(graph.edges).length(0);
+  });
+
+  it('external service - no dependencies created', async () => {
+    const frontend_config = {
+      name: 'architect/frontend',
+      interfaces: {
+        app: 8080
+      },
+      dependencies: {
+        'architect/backend': 'v1'
+      }
+    };
+
+    moxios.stubRequest(`/accounts/architect/services/frontend/versions/v1`, {
+      status: 200,
+      response: { tag: 'v1', config: frontend_config, service: { url: 'architect/frontend:v1' } }
+    });
+
+    const env_config = {
+      services: {
+        'architect/frontend:v1': {
+          interfaces: {
+            app: {
+              host: 'app.localhost',
+              port: 80
+            }
+          }
+        }
+      }
+    };
+
+    mock_fs({
+      '/stack/src/frontend/architect.json': JSON.stringify(frontend_config),
+      '/stack/arc.env.json': JSON.stringify(env_config),
+    });
+
+    const manager = await LocalDependencyManager.createFromPath(axios.create(), '/stack/arc.env.json', undefined, true);
+    const graph = manager.graph;
+    expect(graph.nodes).length(1);
+    expect(graph.nodes[0]).instanceOf(ExternalNode);
+    expect(graph.nodes[0].interfaces.app.host).eq('app.localhost');
+    expect(graph.nodes[0].interfaces.app.port).eq(80);
```

Comment:
Just for clarity, what's the final port mapping result of this? I see we "override" the port from the env config, but is the intent to change the runtime port of the process on the container, or to specify the "external" port that we map to? Seems like both ports are relevant in different context's.

Reply:
Yeah port func as both atm. If its a ServiceNode then we use it as the runtime port for the container. If its an external node than its the port for the external host. Maybe not ideal since most of the time it'll need to be 443. Works great for managed service overrides because often the port is the same in both cases.

Reply:
I believe that means we should do one of two things in a follow up:

1. Ensure that ports are only "overridden" when the host is too via config validation, or
2. Make the exposed port a first-class field that can be overridden separately from the process port

Any thoughts on which you like better?

Reply:
Prob #1